xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/internal/job/MoveJob.java

```diff
@@ -218,6 +222,8 @@ private void move(DocumentReference oldReference, DocumentReference newReference
             if (this.request.isDeleteSource() && this.request.isAutoRedirect()) {
                 this.modelBridge.createRedirect(oldReference, newReference);
             }
+            DocumentMovedEvent newEvent = new DocumentMovingEvent();
```

Comment:
You are supposed to send two events and DocumentMovingEvent is supposed to be sent BEFORE the move as suggested by its name.

Also this event is useless without the references.

The move might fail and you need to make sure you always send an EndEvent to "close" the BeginEvent so you should send the DocumentMovedEvent in the finally and make sure DocumentMovedEvent indicate of it failed or not.